fast_iterator.go

```diff
@@ -0,0 +1,134 @@
+package iavl
+
+import (
+	"errors"
+
+	dbm "github.com/tendermint/tm-db"
+)
+
+var errFastIteratorNilNdbGiven = errors.New("fast iterator must be created with a nodedb but it was nil")
+
+// Iterator is a dbm.Iterator for ImmutableTree
+type FastIterator struct {
+	start, end []byte
+
+	valid bool
+
+	ascending bool
+
+	err error
+
+	ndb *nodeDB
+
+	nextFastNode *FastNode
+
+	fastIterator dbm.Iterator
+}
+
+var _ dbm.Iterator = &FastIterator{}
+
+func NewFastIterator(start, end []byte, ascending bool, ndb *nodeDB) *FastIterator {
+	iter := &FastIterator{
+		start:        start,
+		end:          end,
+		err:          nil,
+		ascending:    ascending,
+		ndb:          ndb,
+		nextFastNode: nil,
+		fastIterator: nil,
+	}
+	iter.Next()
```

Comment:
Why do we do this here, instead of run fast iterator initialization logic here? (Can there be a comment explaining expected behavior)

Reply:
I followed the pattern in [iterator.go](https://github.com/osmosis-labs/iavl/blob/849cd283489e685cdc1efd9bb2b04f58ade6c96f/iterator.go#L171) for consistency.  I guess it is idiomatic with moving the iterator before the first element for setup. I will add a comment for now but if you think that this is confusing, I can refactor them both...

Reply:
A comment definitely suffices!